cloud/deployment/fromfile/fromfile.go

```diff
@@ -158,189 +148,416 @@ func CreateOrUpdate(inputFile, action string, client astro.Client, coreClient as
 				// set default values if none were specified
-				a := workerqueue.SetWorkerQueueValues(listQueues[i].MinWorkerCount, listQueues[i].MaxWorkerCount, listQueues[i].WorkerConcurrency, &listQueues[i], defaultOptions)
-				if deployment.IsDeploymentHosted(deploymentFromFile.Deployment.Configuration.DeploymentType) || deployment.IsDeploymentDedicated(deploymentFromFile.Deployment.Configuration.DeploymentType) {
-					astroMachines := configOptions.AstroMachines
-					for j := range astroMachines {
-						if astroMachines[j].Type == listQueues[i].AstroMachine {
-							astroMachine = astroMachines[j]
-						}
-					}
-					// check if queue is valid
-					err = workerqueue.IsHostedCeleryWorkerQueueInputValid(a, defaultOptions, &astroMachine)
-					if err != nil {
-						return astro.CreateDeploymentInput{}, astro.UpdateDeploymentInput{}, err
-					}
-				} else {
-					// check if queue is valid
-					err = workerqueue.IsCeleryWorkerQueueInputValid(a, defaultOptions)
-					if err != nil {
-						return astro.CreateDeploymentInput{}, astro.UpdateDeploymentInput{}, err
-					}
+				a := workerqueue.SetWorkerQueueValues(listQueues[i].MinWorkerCount, listQueues[i].MaxWorkerCount, listQueues[i].WorkerConcurrency, &workerQueue, defaultOptions)
+				// check if queue is valid
+				if deploymentFromFile.Deployment.Configuration.Executor == deployment.KubeExecutor || deploymentFromFile.Deployment.Configuration.Executor == deployment.KUBERNETES {
+					return errNoUseWorkerQueues
+				}
+				err = workerqueue.IsHostedCeleryWorkerQueueInputValid(a, defaultOptions, &astroMachine)
+				if err != nil {
+					return err
 				}
 				// add it to the list of queues to be created
-				listQueues[i] = *a
-			}
-		} else {
-			// executor is KubernetesExecutor
-			// check if more than one queue is requested
-			if len(listQueues) > 1 {
-				return astro.CreateDeploymentInput{}, astro.UpdateDeploymentInput{},
-					fmt.Errorf("%s %w more than one worker queue. (%d) were requested",
-						deployment.KubeExecutor, workerqueue.ErrNotSupported, len(listQueues))
-			}
-			for i := range listQueues {
-				err = workerqueue.IsKubernetesWorkerQueueInputValid(&listQueues[i])
+				listQueuesRequest = append(listQueuesRequest, *a)
+			} else {
+				var workerQueue astroplatformcore.HybridWorkerQueueRequest
+				workerQueue.Id = &listQueues[i].Id
+				workerQueue.Name = listQueues[i].Name
+				workerQueue.IsDefault = listQueues[i].IsDefault
+				workerQueue.MaxWorkerCount = listQueues[i].MaxWorkerCount
+				workerQueue.MinWorkerCount = listQueues[i].MinWorkerCount
+				workerQueue.WorkerConcurrency = listQueues[i].WorkerConcurrency
+				workerQueue.NodePoolId = *listQueues[i].NodePoolId
+				// check if queue is valid
+				if deploymentFromFile.Deployment.Configuration.Executor == deployment.KubeExecutor || deploymentFromFile.Deployment.Configuration.Executor == deployment.KUBERNETES {
+					return errUseDefaultWokerType
+				}
```

Comment:
maybe we can extract this out to a helper function since it is also used above

Reply:
if deploymentFromFile.Deployment.Configuration.Executor == deployment.KubeExecutor || deploymentFromFile.Deployment.Configuration.Executor == deployment.KUBERNETES { is used in a few places but seems like it would be weird to make it into a it's own function because if statements are quite common